Modules/Sources/Yosemite/Tools/POS/POSCatalogPersistenceService.swift

```diff
@@ -0,0 +1,106 @@
+// periphery:ignore:all
+import Foundation
+import Storage
+
+protocol POSCatalogPersistenceServiceProtocol {
+    /// Clears existing data and persists new catalog data
+    /// - Parameters:
+    ///   - catalog: The catalog to persist
+    ///   - siteID: The site ID to associate the catalog with
+    func replaceAllCatalogData(_ catalog: POSCatalog, siteID: Int64) async throws
+}
+
+final class POSCatalogPersistenceService: POSCatalogPersistenceServiceProtocol {
+    private let grdbManager: GRDBManagerProtocol
+
+    init(grdbManager: GRDBManagerProtocol) {
+        self.grdbManager = grdbManager
+    }
+
+    func replaceAllCatalogData(_ catalog: POSCatalog, siteID: Int64) async throws {
+        DDLogInfo("💾 Persisting catalog with \(catalog.products.count) products and \(catalog.variations.count) variations")
+
+        try await grdbManager.databaseConnection.write { db in
+            DDLogInfo("🗑️ Clearing catalog data for site \(siteID)")
+            // currently, we can't save for more than one site as entity IDs are not namespaced.
+            try PersistedSite.deleteAll(db)
+//            try PersistedSite.deleteOne(db, key: siteID)
```

Comment:
I'm guessing this is related to the need to differentiate products in different sites in p1756965088538469/1756831302.581369-slack-C070SJRA8DP?

nit: can remove the commented out line and add it back when multi-site support is added.
```suggestion
            // currently, we can't save for more than one site as entity IDs are not namespaced.
            try PersistedSite.deleteAll(db)
```

Reply:
> I'm guessing this is related to the need to differentiate products in different sites
Yeah, exactly.